nav2_dwb_controller/dwb_core/src/dwb_local_planner.cpp

```diff
@@ -325,6 +325,9 @@ DWBLocalPlanner::computeVelocityCommands(
 
   prepareGlobalPlan(pose, transformed_plan, goal_pose);
 
+  nav2_costmap_2d::Costmap2D * costmap = costmap_ros_->getCostmap();
+  std::unique_lock<nav2_costmap_2d::Costmap2D::mutex_t> lock(*(costmap->getMutex()));
```

Comment:
This isn't the controller_server, this is a specific plugin. We should have this in the controller_server the way we have it in the planner_server

Reply:
See the nav2_controller package, its pretty much analog to the nav2_planner packagea

Reply:
I put the lock, not in nav2_planner but nav2_navfn_planner because if we put it in nav2_planner it will lock up not only while accessing costmap but also while calculating plans. 

So, I put the lock here not in nav2_controller, considering performance.

Reply:
OK, we can do that. I'll file tickets on other plugins to implement this.